src/main/java/teammates/ui/webapi/CreateAccountAction.java

```diff
@@ -1,59 +1,107 @@
-        } catch (InvalidParametersException e) {
+        } catch (InvalidParametersException ipe) {
             // There should not be any invalid parameter here
-            log.severe("Unexpected error", e);
-            return new JsonResult(e.getMessage(), HttpStatus.SC_INTERNAL_SERVER_ERROR);
+            log.severe("Unexpected error", ipe);
+            return new JsonResult(ipe.getMessage(), HttpStatus.SC_INTERNAL_SERVER_ERROR);
         }
+
         List<InstructorAttributes> instructorList = logic.getInstructorsForCourse(courseId);
-        String joinLink = Config.getFrontEndAppUrl(Const.WebPageURIs.JOIN_PAGE)
-                .withRegistrationKey(instructorList.get(0).getKey())
-                .withEntityType(Const.EntityType.INSTRUCTOR)
-                .toAbsoluteString();
-        EmailWrapper email = emailGenerator.generateNewInstructorAccountJoinEmail(
-                instructorList.get(0).getEmail(), instructorName, joinLink);
-        emailSender.sendEmail(email);
-
-        JoinLinkData output = new JoinLinkData(joinLink);
-        return new JsonResult(output);
+
+        assert !instructorList.isEmpty();
+
+        try {
+            logic.joinCourseForInstructor(instructorList.get(0).getKey(), userInfo.id);
+        } catch (EntityDoesNotExistException ednee) {
+            // All entities should exist in demo course, this exception should not be thrown
+            log.severe("Unexpected error", ednee);
+            return new JsonResult(ednee.getMessage(), HttpStatus.SC_INTERNAL_SERVER_ERROR);
+        } catch (EntityAlreadyExistsException eaee) {
+            // Updated entities should not have conflict with generated entities in new demo course
+            log.severe("Unexpected error", eaee);
+            return new JsonResult(eaee.getMessage(), HttpStatus.SC_INTERNAL_SERVER_ERROR);
+        } catch (InvalidParametersException ipe) {
+            // There should not be any invalid parameter here
+            log.severe("Unexpected error", ipe);
+            return new JsonResult(ipe.getMessage(), HttpStatus.SC_INTERNAL_SERVER_ERROR);
+        }
+
+        try {
+            logic.updateAccountRequest(AccountRequestAttributes
+                    .updateOptionsBuilder(instructorEmail, instructorInstitution)
+                    .withRegisteredAt(Instant.now())
+                    .build());
+        } catch (EntityDoesNotExistException ednee) {
+            // Existence of account request validated before, this exception should not be thrown
+            log.severe("Unexpected error", ednee);
+            return new JsonResult(ednee.getMessage(), HttpStatus.SC_INTERNAL_SERVER_ERROR);
+        } catch (InvalidParametersException ipe) {
+            // There should not be any invalid parameter here
+            log.severe("Unexpected error", ipe);
+            return new JsonResult(ipe.getMessage(), HttpStatus.SC_INTERNAL_SERVER_ERROR);
```

Comment:
The treatment after catch is identical. The catch should therefore be collapsed together i.e. `catch (EntityDoesNotExistException | InvalidParametersException e)`. The explanation comments can always be combined in one paragraph, like:
```java
// EntityDoesNotExistException should not be thrown as existence of account request has been validated before.
// InvalidParametersException should not be thrown as there should not be any invalid parameters.
```

Applies to the previous block as well.

Reply:
ok, will update now.